mp2-v1/src/values_extraction/leaf_receipt.rs

```diff
@@ -0,0 +1,542 @@
+            let address_start = b.add(log_offset, self.add_rel_offset);
+
+            let address_bytes = value.arr.extract_array_large::<_, _, 20>(b, address_start);
+
+            let address_check = address_bytes.equals(b, &self.address);
+            // Extract the signature bytes
+            let sig_start = b.add(log_offset, self.sig_rel_offset);
+
+            let sig_bytes = value.arr.extract_array_large::<_, _, 32>(b, sig_start);
+
+            let sig_check = sig_bytes.equals(b, &self.event_signature);
+
+            // We check to see if the relevant log offset is zero (this indicates a dummy value)
+            let dummy = b.is_equal(log_offset, zero);
+
+            let address_to_enforce = b.select(dummy, t.target, address_check.target);
+            let sig_to_enforce = b.select(dummy, t.target, sig_check.target);
+
+            b.connect(t.target, address_to_enforce);
+            b.connect(t.target, sig_to_enforce);
+
+            for &log_column in self.topics.iter().chain(self.data.iter()) {
+                let data_start = b.add(log_offset, log_column.rel_byte_offset);
+                // The data is always 32 bytes long
+                let data_bytes = value.arr.extract_array_large::<_, _, 32>(b, data_start);
+
+                // Pack the data and get the digest
+                let packed_data = data_bytes.arr.pack(b, Endianness::Big);
+                let data_digest = b.map_to_curve_point(
+                    &std::iter::once(log_column.column_id)
+                        .chain(packed_data)
+                        .collect::<Vec<_>>(),
+                );
+
+                // For each column we use the `column_id` field to tell if its a dummy or not, zero indicates a dummy.
+                let dummy_column = b.is_equal(log_column.column_id, zero);
+
+                let selected_point = b.select_curve_point(dummy_column, curve_zero, data_digest);
+                let selected_point = b.select_curve_point(dummy, curve_zero, selected_point);
+
+                points.push(selected_point);
+            }
+            // If this is a real row we record the gas used in the transaction
+            let gas_select = b.select_curve_point(dummy, curve_zero, gas_digest);
+            points.push(gas_select);
+
+            // We also keep track of which log this is in the receipt to avoid having identical rows in the table in the case
+            // that the event we are tracking can be emitted multiple times in the same transaction but has no topics or data.
+            let log_number = b.constant(F::from_canonical_usize(index + 1));
+            let log_no_digest = b.map_to_curve_point(&[one, log_number]);
+            let log_no_select = b.select_curve_point(dummy, curve_zero, log_no_digest);
+            points.push(log_no_select);
+
+            let increment = b.select(dummy, zero, one);
+            n = b.add(n, increment);
+        }
+
+        (n, b.add_curve_point(&points))
```

Comment:
Actually I think that we will integrate this after generic struct extraction PR, so we should already use the value digest structure expected after that change. In particular, the value digest for a row of the table with columns `id1, ..., idN` and values `v1, ..., vN`, must be computed as `row_id*(D(id1||v1)+...+D(idN||vN)`, where `row_id = H2Int(H(row_unique_data) || num_columns)` and `row_unique_data` is a set of data that uniquely identifies a row in the table (i.e., a primary key). In the case of receipt trie, I think that `row_unique_data` could be `tx_index||log_number||address||event_signature`? It should be sufficient to uniquely identify a row I think. You can look at [these specs](https://www.notion.so/lagrangelabs/Receipt-Trie-12428d1c65a880d484c6d05dccc0dc3f#12428d1c65a880e1902bf086fc80ed08) if you want more details on how to compute the value digest after generic struct extraction.

Reply:
Thanks. I think it would be helpful if you sync this branch with the generic struct extraction one ([this one](https://github.com/Lagrange-Labs/mapreduce-plonky2/pull/404) should be the branch with all the changes from generic struct extraction), so that when you run the integration test you already have the expect DB construction circuits. Up to you whether to do it now or after you did the other changes.